src/Drivers/ClickhouseClient/ClickhouseClientDriver.ts

```diff
@@ -0,0 +1,21 @@
+import {ClickHouseClient} from '@clickhouse/client';
+import {ClickHouseDriver} from './Driver';
+
+export class ClickhouseClientDriver implements ClickHouseDriver<ClickHouseClient> {
+    createClient(client: ClickHouseClient) {
+        return {
+            query(query: string, reqParams: Record<string, unknown>) {
+                return {
+                    async toPromise() {
+                        const cursor = await client.query({
+                            query,
+                            query_params: reqParams
+                        });
+
+                        return cursor.json() as Promise<Object[]>;
+                    }
+                };
```

Comment:
What do you think if we remove `toPromise` method and return Promise immediately? This way we can avoid extra method call and code would look cleaner/shorter. The `query` (e.g. `query<T[]>(...): Promise<T[]>` method can have new generic which would represent returned type.  

Let me know what you think about it?